src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -882,25 +846,38 @@ end
 Get the flow variables of component k
 """
 @inline function get_component(k, u, equations::IdealMhdMultiIonEquations2D)
-  return SVector(u[(k-1)*5+4],
-                 u[(k-1)*5+5],
-                 u[(k-1)*5+6],
-                 u[(k-1)*5+7],
-                 u[(k-1)*5+8])
+  # The first 3 entries of u contain the magnetic field. The following entries contain the density, momentum (3 entries), and energy of each component.
```

Comment:
I think it was explained some time ago, but can you tell me again why the 2D equations have 3 momentum components?

Reply:
The reason is that the cross products in the Lorentz force (and induction equations) are inherently 3D. See, e.g., 
![image](https://user-images.githubusercontent.com/29205060/236817507-f8bee2e3-f86b-43ae-9cde-8350a19b308c.png)
(Derigs et al. eq (2.4), 2018), or
https://github.com/amrueda/Trixi.jl/blob/8662c8f511409e4693c28898e00be31925955ee4/src/equations/ideal_mhd_multiion_2d.jl#L215

There seem to be two strategies to deal with this in the literature:
1. Assume `B1 = B2 = 0`, such that there are no Lorentz forces in `z`.
2. Use 3 components for the magnetic field and also 3 for the momentum.

I selected strategy 2 because it is a bit more general (it can be interpreted as a 2.5D model). BTW, this is the strategy used in the 2D ideal MHD and ideal MHD multicomponent equations of Trixi.

Reply:
Right, I forgot. Thanks for the clarification!